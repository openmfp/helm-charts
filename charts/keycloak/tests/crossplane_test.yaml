suite: crossplane
release:
  name: keycloak
  namespace: openmfp-system
templates:
  - crossplane/*
values:
  - ../test-values.yaml
set:
  crossplane:
    enabled: true
tests:
  - it: matches the snapshot
    asserts:
      - matchSnapshot: {}