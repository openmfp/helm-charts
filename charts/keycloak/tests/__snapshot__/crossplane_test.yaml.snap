matches the snapshot:
  1: |
    apiVersion: openidclient.keycloak.crossplane.io/v1alpha1
    kind: Client
    metadata:
      name: openmfp
    spec:
      forProvider:
        accessType: CONFIDENTIAL
        clientId: openmfp
        enabled: true
        name: OpenMFP
        realmIdRef:
          name: openmfp
        standardFlowEnabled: true
        validRedirectUris:
          - http://localhost:8000/callback*
          - http://localhost:4300/callback*
      providerConfigRef:
        name: keycloak-provider-config
      writeConnectionSecretToRef:
        name: portal-client-secret-openmfp
        namespace: openmfp-system
  2: |
    apiVersion: openidclient.keycloak.crossplane.io/v1alpha1
    kind: ClientDefaultScopes
    metadata:
      name: openmfp-default-scopes
    spec:
      forProvider:
        clientIdRef:
          name: openmfp
        defaultScopes:
          - profile
          - email
          - basic
          - acr
          - groups
          - trusted-openmfp-auds
        realmIdRef:
          name: openmfp
      providerConfigRef:
        name: keycloak-provider-config
  3: |
    apiVersion: defaults.keycloak.crossplane.io/v1alpha1
    kind: DefaultGroups
    metadata:
      name: default
    spec:
      forProvider:
        groupIdsRefs:
          - name: portal
        realmIdRef:
          name: openmfp
      providerConfigRef:
        name: keycloak-provider-config
  4: |
    apiVersion: group.keycloak.crossplane.io/v1alpha1
    kind: Group
    metadata:
      name: portal
    spec:
      forProvider:
        name: portal
        realmIdRef:
          name: openmfp
      providerConfigRef:
        name: keycloak-provider-config
  5: |
    apiVersion: openidclient.keycloak.crossplane.io/v1alpha1
    kind: ClientScope
    metadata:
      name: groups-client-scope
    spec:
      forProvider:
        description: When requested, this scope will map a user's group memberships to a claim
        includeInTokenScope: true
        name: groups
        realmIdRef:
          name: openmfp
      providerConfigRef:
        name: keycloak-provider-config
  6: |
    apiVersion: openidgroup.keycloak.crossplane.io/v1alpha1
    kind: GroupMembershipProtocolMapper
    metadata:
      name: group-membership-mapper
    spec:
      forProvider:
        claimName: groups
        clientIdRef:
          name: groups-client-scope
        name: group-membership-mapper
        realmIdRef:
          name: openmfp
      providerConfigRef:
        name: keycloak-provider-config
  7: |
    apiVersion: openidclient.keycloak.crossplane.io/v1alpha1
    kind: ClientScope
    metadata:
      name: trusted-openmfp-audiences
    spec:
      forProvider:
        includeInTokenScope: true
        name: trusted-openmfp-audiences
        realmIdRef:
          name: openmfp
      providerConfigRef:
        name: keycloak-provider-config
  8: |
    apiVersion: keycloak.crossplane.io/v1beta1
    kind: ProviderConfig
    metadata:
      name: keycloak-provider-config
      namespace: openmfp-system
    spec:
      credentials:
        secretRef:
          key: config
          name: keycloak-provider-config
          namespace: openmfp-system
        source: Secret
  9: |
    apiVersion: v1
    kind: Secret
    metadata:
      name: keycloak-provider-config
      namespace: openmfp-system
    stringData:
      config: |-
        {
          "client_id":"admin-cli",
          "username": "keycloak-admin",
          "password": "admin",
          "url": "http://keycloak-http.openmfp-system.svc.cluster.local:8080",
          "realm": "master"
        }
  10: |
    apiVersion: realm.keycloak.crossplane.io/v1alpha1
    kind: Realm
    metadata:
      name: openmfp
    spec:
      forProvider:
        accessTokenLifespan: 1h
        attributes:
          organizationsEnabled: "true"
        displayName: OpenMFP
        displayNameHtml: <b>OpenMFP</b>
        enabled: true
        loginWithEmailAllowed: true
        realm: openmfp
        registrationAllowed: false
        registrationEmailAsUsername: true
      providerConfigRef:
        name: keycloak-provider-config
